vtm-android/src/org/oscim/android/tiling/MBTilesBitmapTileDataSource.java

```diff
@@ -0,0 +1,281 @@
+    }
+
+    public int getMaxZoom() throws Exception {
+        checkMetadata();
+        String maxZoomStr = metadataMap.get("maxzoom");
+        if (maxZoomStr != null) {
+            return Integer.parseInt(maxZoomStr);
+        }
+        return -1;
+    }
+
+    public String getName() throws Exception {
+        checkMetadata();
+        return metadataMap.get("name");
+    }
+
+    /**
+     * Get a Tile's image bytes from the database.
+     *
+     * @param tx    the x tile index.
+     * @param tyOsm the y tile index, the osm way.
+     * @param zoom  the zoom level.
+     * @return the tile image bytes.
+     * @throws Exception
+     */
+    public byte[] getTile(int tx, int tyOsm, int zoom) throws Exception {
+        int ty = tyOsm;
+        int[] tmsTileXY = osmTile2TmsTile(tx, tyOsm, zoom);
+        ty = tmsTileXY[1];
+
+
+        Cursor cursor = null;
+        try {
+            cursor = mDb.rawQuery(SELECTQUERY, new String[]{String.valueOf(zoom), String.valueOf(tx), String.valueOf(ty)});
+            if (cursor.moveToFirst()) {
+                return cursor.getBlob(0);
+            }
+        } finally {
+            if (cursor != null)
+                cursor.close();
+        }
+        return null;
+    }
+
+    public String getVersion() throws Exception {
+        checkMetadata();
+        return metadataMap.get("version");
+    }
+
+    private static android.graphics.Bitmap makeBitmapTransparent(android.graphics.Bitmap originalBitmap, int alpha) {
+        android.graphics.Bitmap newBitmap = android.graphics.Bitmap.createBitmap(originalBitmap.getWidth(), originalBitmap.getHeight(), android.graphics.Bitmap.Config.ARGB_8888);
+        Canvas canvas = new Canvas(newBitmap);
+        Paint alphaPaint = new Paint();
+        alphaPaint.setAlpha(alpha);
+        canvas.drawBitmap(originalBitmap, 0, 0, alphaPaint);
+        return newBitmap;
+    }
+
+    private static android.graphics.Bitmap makeTransparent(android.graphics.Bitmap bit, int colorToRemove) {
+        int width = bit.getWidth();
+        int height = bit.getHeight();
+        android.graphics.Bitmap myBitmap = android.graphics.Bitmap.createBitmap(width, height, android.graphics.Bitmap.Config.ARGB_8888);
+        int[] allpixels = new int[myBitmap.getHeight() * myBitmap.getWidth()];
+        bit.getPixels(allpixels, 0, myBitmap.getWidth(), 0, 0, myBitmap.getWidth(), myBitmap.getHeight());
+        myBitmap.setPixels(allpixels, 0, width, 0, 0, width, height);
+
+        for (int i = 0; i < myBitmap.getHeight() * myBitmap.getWidth(); i++) {
+            if (allpixels[i] == colorToRemove)
+                allpixels[i] = Color.alpha(Color.TRANSPARENT);
+        }
+
+        myBitmap.setPixels(allpixels, 0, myBitmap.getWidth(), 0, 0, myBitmap.getWidth(), myBitmap.getHeight());
+        return myBitmap;
+    }
+
+    private static int[] osmTile2TmsTile(int tx, int ty, int zoom) {
+        return new int[]{tx, (int) ((Math.pow(2, zoom) - 1) - ty)};
```

Comment:
Can move the method in `MercatorProjection` utilities and avoid the array creation for each conversion for better performance, since just need the y coordinate.